src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,925 @@
+    EXPECT_EQ(OK, status);
+
+    // Create the parmeterization to contain the desired number of segments
+    CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_param,
+                                             TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS);
+
+    // Get all of the points for the arc length parameterization (Not constant arc length
+    // segments)
+    shared_polynomial_getArcLengthParametrizationOrder3(
+        path, path_parameters.t_start, path_parameters.t_end, arc_length_param);
+
+    std::vector<double> velocity     = getSpeedsFromTrajectory(path_parameters);
+    std::vector<double> acceleration = getAccelerationsFromSpeed(velocity, &trajectory);
+
+    for (float vel : velocity)
+    {
+        EXPECT_TRUE(vel <= path_parameters.max_allowable_speed * 1.1 &&
+                    vel >= path_parameters.initial_speed);
+    }
+
+    EXPECT_NEAR(velocity.back(), path_parameters.final_speed, 0.1);
+    EXPECT_FLOAT_EQ(velocity.front(), path_parameters.initial_speed);
+
+    // Loop through the positive acceleration of the bang-bang profile
+    unsigned int i = 0;
+    while (acceleration[i] > path_parameters.max_allowable_acceleration * 0.95)
+    {
+        EXPECT_NEAR(path_parameters.max_allowable_acceleration, acceleration[i], 0.01);
+        i++;
+    }
+    i++;  // Skip the transition acceleration from positive to negative
+
+    // Loop through the negative acceleration portion of the bang-bang profile
+    while (acceleration[i] < -path_parameters.max_allowable_acceleration * 0.95)
+    {
+        EXPECT_NEAR(-path_parameters.max_allowable_acceleration, acceleration[i], 0.01);
+        i++;
+    }
+
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].position.x,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_end).x,
+        0.01);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].position.y,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_end).y,
+        0.01);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.x,
+                shared_polynomial2d_getValueOrder3(path, path_parameters.t_start).x,
+                0.01);
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.y,
+                shared_polynomial2d_getValueOrder3(path, path_parameters.t_start).y,
+                0.01);
+}
+
+TEST_F(TrajectoryPlannerTest, check_trajectory_path_length_reverse_parameterization)
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {0, 0, 1, 0}},
+        .y = {.coefficients = {0, 0, 1, 0}},
+
+    };
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path                       = path;
+    path_parameters.t_start                    = 1;
+    path_parameters.t_end                      = 0;
+    path_parameters.num_segments               = 5999;
```

Comment:
Can we change this to TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS - 1? (same for all the other tests that use 5999)

Reply:
Changed all references to 5999 👍